sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -685,6 +698,11 @@ def _request(
         # type: (...) -> HttpRequest
         """Create HttpRequest object.
 
+        If content is not None, guesses will be used to set the right body:
+        - If content is an XML tree, will serialize as XML
+        - If content-type starts by "text/", set the content as text
+        - Else, try JSON serialization
+
```

Comment:
Why not put If content-type starts by "text/", set the content as text first?

Reply:
Because "text/xml" is actually a thing, and receiving a XML tree the only thing we can do is serialize it. If I do the opposite, I might receive "text/xml" + an XML tree and that will try to send a XML tree to "requests" (bad thing would happen :))